pkg/serverless/proxy/transport.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package proxy
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type proxyTransport struct {
+	currentInvocationDetails *invocationDetails
+	processor                invocationProcessor
+}
+
+func (p *proxyTransport) RoundTrip(request *http.Request) (*http.Response, error) {
+	log.Debug("[proxy] new request to %s", request.URL)
+
+	// enrich the currentInvocationDetails object
+	requestProcessor(p.currentInvocationDetails, request)
+	if p.currentInvocationDetails.isComplete() {
+		p.processor.process(p.currentInvocationDetails)
```

Comment:
I think it's a bit confusing to have a function called `requestProcessor` in this file, but also call `p.processor.process()`. Can we differentiate the names of these functions so it's more obvious which function does what?

Reply:
yeah I've renamed it to `enrichCurrentInvocation`